### DISCUSSION TITLE
Porting explicitly to ESP32-CAM (driver support for the camera)

### DISCUSSION THREAD

**Discussion by ghost** [micropython/micropython#9692] [Hardware & Peripherals]
I'm having trouble getting MicroPython working on an ESP32-CAM board with the OV2640 camera and SD card slot on board, and powering with the ESP32-CAM-MB power/USB daughter board.  Not only can I not get it to work (it flashes but cannot boot), but I can't seem to follow any guide (there seem to be none after 2019) on how to compile with support for the camera.  The stock binary images aren't working for me either.

Has this been done so much that nobody talks about it any more?  When searching for similar discussions, I can't find any mention of the ESP32-CAM board on these (or the archive) forums at all.

I was thinking that little $9 intelligent WiFi cameras would be pretty popular, and I'm just not that bright... is there something I need to know about these?  Are they poisonous to children or something?

Thank for any input whatsoever, really!!

**Comment by scruss** [comment]
This would be useful to me, too, as I've tried and failed any build instructions I can find

**Comment by jimmo** [comment]
I don't know the answer, but I did hear that CircuitPython were putting a lot of effort into camera support on ESP32, so it might be worth checking there?
